tools/snapshot/internal/export/export.go

```diff
@@ -0,0 +1,245 @@
+	grpcUser           = "clio-snapshot"
+	markerNum          = 16
+)
+
+type GRPCClient struct {
+	Client pb.XRPLedgerAPIServiceClient
+	conn   *grpc.ClientConn
+}
+
+func (c *GRPCClient) Close() error {
+	return c.conn.Close()
+}
+
+func createGRPCClient(serverAddr string) (*GRPCClient, error) {
+	opts := []grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())}
+
+	conn, err := grpc.NewClient(serverAddr, opts...)
+	if err != nil {
+		return nil, fmt.Errorf("Failed to dial: %v", err)
+	}
+
+	client := pb.NewXRPLedgerAPIServiceClient(conn)
+	return &GRPCClient{
+		Client: client,
+		conn:   conn,
+	}, nil
+}
+
+func getLedgerDeltaData(client pb.XRPLedgerAPIServiceClient, seq uint32, path string) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	request := pb.GetLedgerRequest{}
+	ledger := &pb.LedgerSpecifier{
+		Ledger: &pb.LedgerSpecifier_Sequence{
+			Sequence: seq,
+		},
+	}
+	request.Ledger = ledger
+	request.User = grpcUser
+	request.GetObjectNeighbors = true
+	request.Transactions = true
+	request.Expand = true
+	request.GetObjects = true
+
+	response, err := client.GetLedger(ctx, &request)
+
+	if err != nil {
+		log.Fatalf("Error getting ledger data: %v", err)
+	}
+
+	saveLedgerDeltaData(seq, response, path)
+
+	log.Printf("Processing delta sequence: %d\n", seq)
+}
+
+func roundDown(n uint32, roundTo uint32) uint32 {
+	return n - (n % roundTo)
+}
+
+func saveLedgerDeltaData(seq uint32, response *pb.GetLedgerResponse, path string) {
+	subPath := filepath.Join(path, fmt.Sprintf("ledger_diff_%d", roundDown(seq, deltaDataFolderDiv)))
+	err := os.MkdirAll(subPath, os.ModePerm)
+	if err != nil {
+		log.Fatalf("Error creating directory: %v", err)
+	}
+
+	protoData, err := proto.Marshal(response)
+	if err != nil {
+		log.Fatalf("Error marshalling data: %v", err)
+	}
+
+	filePath := filepath.Join(subPath, fmt.Sprintf("%d.dat", seq))
+
+	err = os.WriteFile(filePath, protoData, 0644)
+	if err != nil {
+		log.Fatalf("failed to write file: %v", err)
+	}
+}
+
+func generate256Markers(markerNum uint32) [][32]byte {
+	var byteArray [32]byte
+
+	incr := 256 / markerNum
+
+	var byteArrayList [][32]byte
+
+	for i := 0; i < int(markerNum); i++ {
+		byteArray[0] = byte(i * int(incr)) // Increment the highest byte
+		byteArrayList = append(byteArrayList, byteArray)
+	}
+
+	return byteArrayList
+}
+
+func saveLedgerData(client pb.XRPLedgerAPIServiceClient, seq uint32, marker []byte, end []byte, path string) {
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	request := pb.GetLedgerDataRequest{}
+	ledger := &pb.LedgerSpecifier{
+		Ledger: &pb.LedgerSpecifier_Sequence{
+			Sequence: seq,
+		},
+	}
+	request.Ledger = ledger
+	request.Marker = marker[:]
+	if end != nil {
+		request.EndMarker = end[:]
+	}
+	request.User = grpcUser
+
+	subPath := filepath.Join(path, fmt.Sprintf("ledger_data_%d", seq), fmt.Sprintf("marker_%x", marker))
+	err := os.MkdirAll(subPath, os.ModePerm)
```

Comment:
Will this fail if the directory already exists?

Reply:
In this case, it does nothing. No error.